File: Programmers/Lv3/Lv3_등굣길.cpp
﻿#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(int m, int n, vector<vector<int>> puddles) {
	int road[101][101] = { 0, };
	int check[101][101] = { 0, };

	for (int i = 0; i < puddles.size(); i++)			// 물은 0 
		check[puddles[i][1]][puddles[i][0]] = -1;

	road[1][0] = 1;
	for (int i = 1; i <= n; i++) {
		for (int j = 1; j <= m; j++) {
			if (check[i][j] == -1)						// 물
				road[i][j] = 0;
			else										// 땅
				road[i][j] = (road[i][j - 1] + road[i - 1][j]) % 1000000007;
		}
	}
	return road[n][m];
}

File: Programmers/Lv3/Lv3_등굣길.py
def solution(m, n, puddles):
    road = [[0 for col in range(m+1)] for row in range(n+1)]
    check = [[0 for col in range(m+1)] for row in range(n+1)]

    for i in puddles:
        check[i[1]][i[0]] = -1

    road[1][0] = 1
    for i in range(1, n+1):
        for j in range(1, m+1):
            if check[i][j] == -1:
                road[i][0] = 0
            else:
                road[i][j] = (road[i][j-1] + road[i-1][j]) % 1000000007

    return road[n][m]

print(solution(4, 3, [[2, 2]]))